src/Algebra/Graph/NonEmpty.hs

```diff
@@ -377,8 +395,31 @@ hasVertex v = foldg1 (==v) (||) (||)
 -- hasEdge x y . 'removeEdge' x y == const False
 -- hasEdge x y                  == 'elem' (x,y) . 'edgeList'
 -- @
+{-# SPECIALISE hasEdge :: Int -> Int -> NonEmptyGraph Int -> Bool #-}
 hasEdge :: Eq a => a -> a -> NonEmptyGraph a -> Bool
-hasEdge = T.hasEdge
```

Comment:
Since you now provide much faster implementations compared to the default ones in `ToGraph`, you should use them in the `instance ToGraph` declaration.

Reply:
Writing this improvement in the `ToGraph` instance drop a bit the performances:

I think this is because I can't use `NonEmpty` graphs in `ToGraph`, so I can 't use `Algebra.Graph.NonEmpty.Vertex` but only `Algebra.Graph.Vertex`, and thus, in the final step, I can only use `foldg` instead of `foldg1`.

Reply:
@nobrakal I meant this:

```haskell
instance ToGraph NonEmpty where
    ...
    hasEdge = hasEdge -- use faster implementation instead of the default one
```

Surely this can't reduce the performace of `hasEdge` itself!

Reply:
But users of the typeclass might get the benefit.

So, the rule is: if you use the default implementation, i.e. `x = T.x` then the instance declaration uses default methods. But otherwise you need to redefine instance methods, i.e. `x = x`.

Reply:
Oh yes, it is done now :)